elastic/kibana
# Improve Clean Repository error message.

I updated the error-handling and display logic to surface only the pertinent details of the error instead of rendering the raw JSON response from ES (originally introduced in https://github.com/elastic/kibana/pull/53047). I also updated the tests.

Here's how this error looks now:

![image](https://user-images.githubusercontent.com/1238659/140823622-b68cf5a6-a8f9-491c-9b1d-71378262076e.png)

## Steps to test

1. Follow the steps from the Snapshot and Restore README to set up a repository
2. Create a policy that will capture your entire cluster (the default)
3. Install the logs sample data and then duplicate it using the requests below, in Console. This will increase the size of the snapshot you'll delete later, so that deleting it takes a few seconds.
4. Execute the policy to take a snapshot
5. In one tab, open the details of the repository and in another table open the list of snapshots
6. Delete the snapshot and immediately click the "Clean up repository" button in the other tab

The repository will fail to clean up because it's in the process of deleting a snapshot and you'll see the error above.

```
# Clone the sample data a bunch of times
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-1
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-2
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-3
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-4
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-5
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-6
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-7
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-8
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-9
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-10
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-12
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-13
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-14
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-15
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-16
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-17
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-18
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-19
POST /kibana_sample_data_logs/_clone/kibana_sample_data_logs-20
```
